### PR TITLE
FINERACT-809 - Upgrade rest-assured

### PR DESCRIPTION
**Pull request by xurror**
## Description
Upgrade rest-assured to latest spring boot compatible version https://issues.apache.org/jira/browse/FINERACT-809

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [x] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [ ] API documentation at https://github.com/apache/fineract/blob/develop/api-docs/apiLive.htm has been updated with details of any API changes.

- [ ] Integration tests have been created/updated for verifying the changes made.

- [x] All Integrations tests are passing with the new commits.

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

### CODE REVIEW

**Comment by xurror** [comment]
This PR resolves the issue with https://github.com/apache/fineract/pull/677
@vorburger and @awasum please can you review this.

**Comment by awasum** [comment]
Something is failing.

**Comment by xurror** [comment]
This is the 2nd PR I've sent and Travis keeps failing, all tests work
locally for the 2 PRs.

I've also checked other PRs with failed travis CI and the error seems to be
thesame.

Perhaps we need to troubleshoot that.

**Comment by awasum** [comment]
org.apache.fineract.integrationtests.ClientSavingsIntegrationTest > testSavingsAccount_DormancyTracking FAILED
    java.lang.ArrayIndexOutOfBoundsException at ClientSavingsIntegrationTest.java:1977

Please go see the travis log and see 1 integration test failing.

See line number: https://travis-ci.org/apache/fineract/builds/657603142#L3638

you can always debug or troubleshoot yourself.its easy.

**Comment by xurror** [comment]
it's OK now, not sure what was wrong before but it got resolved by itself

File: fineract-provider/dev-dependencies.gradle
Comment (vorburger) [inline fineract-provider/dev-dependencies.gradle]: Duplicate of line 110, remove this

File: fineract-provider/build.gradle
Comment (vorburger) [inline fineract-provider/build.gradle]: Just completely remove instead of commenting this out, please

File: fineract-provider/dependencies.gradle
Comment (vorburger) [inline fineract-provider/dependencies.gradle]: Please remove instead

**Review by vorburger — CHANGES_REQUESTED** [review]
Nice!! Congratulations.

Very minor clean up suggested, then I'm happy to merge.

**Comment by awasum** [comment]
This still failing...Check the Travis logs.

**Comment by vorburger** [comment]
> This still failing...Check the Travis logs.

not sure if related to this PR, or flaky general problem... see https://issues.apache.org/jira/browse/FINERACT-852, needs more investigation. @xurror do you want to have a closer look at what could be causing those ArrayIndexOutOfBoundsException, and comment on that JIRA? Or just rebase and retry building this to see if it goes away?

**Comment by xurror** [comment]
Ok, checking this

**Comment by xurror** [comment]
I ran. /gradlew clean build, clean check and clean integrationtest on this locally and they all passed but travus still fails with duplicate values query exceptions and unable to execute queries.

And I can recall before @vorburger reviewed and requested clean up changes, it had passed travis test. The changes I made were just to remove commented dependencies and a duplicate import. 
I think the rest-assured update is but the failing test issue persists. We'll, that's my opinion

**Comment by vorburger** [comment]
For the record, the rebase conflicts which now appear here (after other merges) are fixed in #726.

We'll see if the build passes there (probably still not, we'll have to fix those integration tests).

**Comment by xurror** [comment]
@vorburger, @percyashu There seems to be a rest-assured related problem here.
Test fails from this definition in all the test packages
`ResquestSpecification` requestSpec = new RequestSpecBuilder().setContentType(ContentType.JSON).build()`
Logs report of an AbstractMethodError. Not sure where thats coming from cause all the methods used are in the docs and are used as described.

**Comment by awasum** [comment]
@percyashu or @xurror ..Please can we look into this?

**Comment by xurror** [comment]
Sure

**Comment by xurror** [comment]
From what I read [here](https://github.com/rest-assured/rest-assured/issues/1175#issuecomment-499593018), with java8 we can only use atmost version 3.3.0 of restassured. So far this is an upgrade but to upgrade to the latest version, we first need to upgrade to java9 or higher. Well that's my opinion.

@awasum, do you buy this idea?

**Comment by awasum** [comment]
This is now passing? What did we fix?

**Comment by awasum** [comment]
> From what I read [here](https://github.com/rest-assured/rest-assured/issues/1175#issuecomment-499593018), with java8 we can only use atmost version 3.3.0 of restassured. So far this is an upgrade but to upgrade to the latest version, we first need to upgrade to java9 or higher. Well that's my opinion.
> 
> @awasum, do you buy this idea?

Do it. Did u already add the fix?

**Review by vorburger — APPROVED** [review]

**Comment by vorburger** [comment]
> This is now passing? What did we fix?

I suspect that it's "flaky" - passes sometimes, but fails. The question is if with the rest-assured upgrade Integration Tests fails a lot more often than they used to before the upgrade, or if that was a total coincidence...

Given that we have it passing how, I would say why don't we merge this? If we suddenly see ITs are failing a lot more often, we could always revert this.

>> @awasum, do you buy this idea?
> Do it. Did u already add the fix?

Note [FINERACT-846](https://issues.apache.org/jira/browse/FINERACT-846) re. Java 11; I would to do that separately than this PR.

**Comment by awasum** [comment]
> > This is now passing? What did we fix?
> 
> I suspect that it's "flaky" - passes sometimes, but fails. The question is if with the rest-assured upgrade Integration Tests fails a lot more often than they used to before the upgrade, or if that was a total coincidence...
> 
> Given that we have it passing how, I would say why don't we merge this? If we suddenly see ITs are failing a lot more often, we could always revert this.
> 
> > > @awasum, do you buy this idea?
> > > Do it. Did u already add the fix?
> 
> Note [FINERACT-846](https://issues.apache.org/jira/browse/FINERACT-846) re. Java 11; I would to do that separately than this PR.

Seems @xurror  forced pushed this 3 hours ago: https://github.com/apache/fineract/pull/719/commits/627c7032565d31aa5eb6d75eac4f35a1c6310319, why the thing passed.

**Comment by awasum** [comment]
Lets merged this....Now...

**Comment by xurror** [comment]
> > From what I read [here](https://github.com/rest-assured/rest-assured/issues/1175#issuecomment-499593018), with java8 we can only use atmost version 3.3.0 of restassured. So far this is an upgrade but to upgrade to the latest version, we first need to upgrade to java9 or higher. Well that's my opinion.
> > @awasum, do you buy this idea?
> 
> Do it. Did u already add the fix?

yes

**Comment by xurror** [comment]
Just a quick reminder. In my last push, I just regressed io.restassured to an older more compatible version.

**Pull request merged**
